src/_reducers/SignupReducers.js

```diff
@@ -0,0 +1,28 @@
+import { fromJS } from 'immutable';
+import { SIGNUP_FAILED, SIGNUP_FIELD_UPDATE, SIGNUP_START } from '../_constants/ActionTypes';
+
+const initialState = fromJS({
+    validatedOnce: false,
+    residence: '',
+    password: '',
+    email: '',
+    confirmPassword: '',
+    progress: false,
+});
+
+export default (state = initialState, action) => {
+    switch (action.type) {
+        case SIGNUP_START: {
+            return state.set('progress', true);
+        }
+        case SIGNUP_FIELD_UPDATE: {
+            return state.set(action.fieldName, action.fieldValue);
+        }
+        case SIGNUP_FAILED: {
```

Comment:
Sure, will change it, about the noun vs verb, it have been loosely named, but self-explanatory, this remind me of the hardest problem in programming haha